### PR TITLE
test: legible output of `std::chrono` values

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10180)

### CODE REVIEW

File: google/cloud/internal/type_traits.h [hunk 1]
```diff
@@ -16,6 +16,9 @@
 #define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_TYPE_TRAITS_H
 
 #include "google/cloud/version.h"
+#include <ostream>
```
Comment (coryan) [inline google/cloud/internal/type_traits.h:19]: Probably `#include <iosfwd>` would do in this case.

File: google/cloud/testing_util/chrono_matchers.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Provide `std::chrono::duration` and `std::chrono::time_point` output
+ * streaming when otherwise unsupported. This enables the printing of
+ * legible chrono values from googletest, rather than raw bytes, when a
+ * match fails.
+ *
+ * It is strictly undefined behavior to add to any `std` namespace, but
+ * it works, and, in this test-only case, is well worth it.
+ */
+namespace std {
```
Comment (coryan) [inline google/cloud/testing_util/chrono_matchers.h:53]: If you are willing to open the namespace, then adding `PrintTo()` overloads is also an option:

https://github.com/google/googletest/blob/main/docs/advanced.md#teaching-googletest-how-to-print-your-values

Unless you worry about a `PrintTo()` already in the `std::chrono()` namespace

File: google/cloud/internal/type_traits.h [hunk 3]
```diff
@@ -37,6 +40,23 @@ namespace internal {
 template <typename...>
 using void_t = void;
 
+/**
```
Comment (coryan) [inline google/cloud/internal/type_traits.h:43]: Alternatively:

```cc
template <typename T, typename AlwaysVoid = void>
class IsOStreamable : public std::false_type {};

template <typename T>
class IsOStreamable<T, void_t<decltype(std::declval<std::ostream&>() << std::declval<U>())>> : public std::true_type {};
```

https://godbolt.org/z/oPqW6e4nq

**Review by coryan — APPROVED** [review]
LGTM, and this seems easier to adopt through all tests than my proposal.
A couple of (purely optional) comments.

File: google/cloud/internal/type_traits.h (hunk 3 above)
Comment (devbww) [reply google/cloud/internal/type_traits.h:44]: Done.

File: google/cloud/testing_util/chrono_matchers.h (hunk 2 above)
Comment (devbww) [reply google/cloud/testing_util/chrono_matchers.h:53]: The win with the existing incantation is that normal, outside-googletest streaming also works.  I admit that that whole thing is currently wrapped in "chrono_matcher" clothing, but having it also for non-googletest tests is nice.  We could change the filename.

File: google/cloud/internal/type_traits.h (hunk 1 above)
Comment (devbww) [reply google/cloud/internal/type_traits.h]: Done.

File: google/cloud/testing_util/chrono_matchers.h (hunk 2 above)
Comment (coryan) [reply google/cloud/testing_util/chrono_matchers.h:53]: Ack.  Your call I think.

**Review by coryan — APPROVED** [review]

File: google/cloud/testing_util/chrono_matchers.h (hunk 2 above)
Comment (devbww) [reply google/cloud/testing_util/chrono_matchers.h]: Renamed to "chrono_output".

**Pull request merged**